modules/@apostrophecms/schema/ui/apos/components/AposInputRelationship.vue

```diff
@@ -76,8 +76,28 @@ export default {
       return apos.modules[this.field.withType].components.managerModal;
     }
   },
-  mounted() {
+  async mounted() {
     this.validateAndEmit();
+
+    // Check if this is joined to an image.
+    if (this.field.withType === '@apostrophecms/image' && this.value.data && this.value.data.length > 0) {
+      // If it is, go get the URLs and populate them on `value`.
+      let i = 0;
+      for (const image of this.value.data) {
+        if (image.attachment._urls) {
```

Comment:
This should always already be present in whatever API response populates the join in the first place. A new fetch should not be necessary, are you sure the URLs are not populating?

Reply:
It isn't present because the image object that's already here is what comes directly from the database. _urls is not populated there.

Reply:
It should. Hmm. Let me peep.